RecoTracker/LSTCore/interface/alpaka/LSTESData.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RecoTracker_LSTCore_interface_alpaka_LSTESData_h
+#define RecoTracker_LSTCore_interface_alpaka_LSTESData_h
+
+#include "RecoTracker/LSTCore/interface/alpaka/Constants.h"
+#include "RecoTracker/LSTCore/interface/alpaka/EndcapGeometryBuffer.h"
+#include "RecoTracker/LSTCore/interface/alpaka/Module.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+
+#include <filesystem>
+#include <memory>
+
+namespace SDL {
+
+  struct pixelMap;
+
+  template <typename TDev>
+  struct LSTESData {
+    uint16_t nModules;
+    uint16_t nLowerModules;
+    unsigned int nPixels;
+    unsigned int nEndCapMap;
+    std::shared_ptr<const modulesBuffer<TDev>> modulesBuffers;
+    std::shared_ptr<const EndcapGeometryBuffer<TDev>> endcapGeometryBuffers;
+    std::shared_ptr<const pixelMap> pixelMapping;
```

Comment:
For `*Buffer` it makes sense to use directly by value since the data is device-specific. But for `pixelMap` wouldn't it make more sense to keep it as `shared_ptr` so that even when using multiple devices there is still only one copy of the data?

Reply:
> But for `pixelMap` wouldn't it make more sense to keep it as `shared_ptr` so that even when using multiple devices there is still only one copy of the data?

Poking around more, I see now that the `pixelMapping` is indeed not copied to the device in `CopyToDevice<SDL::LSTESData<alpaka_common::DevHost>>::copyAsync()`. Is the `pixelMapping` then used only in host code? If yes, I agree sharing the `pixelMap` is better than copying it around.

But I'm also concerned that having one ES data product that contains both device and host side data could be confusing (it certainly confused me). At minimum it would be good to make it more clear which members are on the device memory and which are on the host memory. I'd also suggest to consider moving all data that are used only in host code into a separate ES data product (then you also wouldn't have to care about those in `CopyToDevice<SDL::LSTESData<alpaka_common::DevHost>>::copyAsync()`).